src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBResource.cs

```diff
@@ -136,7 +136,14 @@ internal ReferenceExpression GetConnectionString(string? databaseName = null, st
 
         if (IsEmulator || UseAccessKeyAuthentication)
         {
-            builder.AppendFormatted(ConnectionStringExpression);
+            if (databaseName is null)
+            {
+                builder.Append($"{new BicepSecretOutputReference(GetKeyValueSecretName(databaseName, containerName), this)}");
```

Comment:
This should be using the new keyvault reference. The new format is connectionstrings--{resourcename}. Don't add any hierearchy, just use resource names.